XandO_table.java

```diff
@@ -25,23 +25,30 @@ else if (checkForDraw()) {
         return -1; // return code for input error
     }
 
-    private byte validMove(char val) {
+    public static byte validChar(char val) {
         if (val == 'X' || val == 'x') {
             return 1;
         }
         else if (val == 'O' || val == 'o') {
-            return 0;
+            return 2;
         }
         return -1;
     }
 
     private boolean updateCell(int row, int col, byte val) {
         if (row >= 0 && row < ROW_SIZE && col >= 0 && col < COLUMN_SIZE) {
+            if (!isCellOccupied(row, col)) {
+                return false; // If a move has already been played on this cell then return false
+            }
             tableXO[row][col] = val;
             moves++;
             return true;
         }
-        return false;
+        return false; // If row or col out of range, fail
+    }
+
+    private boolean isCellOccupied(int row, int col) {
+        return tableXO[row][col] == 0;
```

Comment:
tableX0[row][col] has values X and 0s ? If so what happens if tableX0 returns an X ?